apps/explorer/lib/explorer/graphql.ex

```diff
@@ -232,9 +232,57 @@ defmodule Explorer.GraphQL do
     )
   end
 
-  def token_txtransfers_query_for_address(address_hash) do
-    token_txtransfers_query()
-    |> where([t], t.to_address_hash == ^address_hash or t.from_address_hash == ^address_hash)
+  def token_txtransfers_query_for_address(address_hash, first) do
+    tt_limit = first * 2
+
+    tokens =
+      from(
+        tt in TokenTransfer,
+        where: not is_nil(tt.transaction_hash),
+        where: tt.to_address_hash == ^address_hash,
+        or_where: tt.from_address_hash == ^address_hash,
+        select: %{
+          transaction_hash: tt.transaction_hash,
+          block_number: tt.block_number,
+          to_address_hash: tt.to_address_hash,
+          from_address_hash: tt.from_address_hash
+        },
+        distinct: [desc: tt.block_number, desc: tt.transaction_hash],
+        order_by: [
+          desc: tt.block_number,
+          desc: tt.transaction_hash,
+          desc: tt.from_address_hash,
+          desc: tt.to_address_hash
+        ],
+        limit: ^tt_limit,
+        offset: 0
+      )
+
+    from(
```

Comment:
Looks like you missed the order by clause from the original query
```
ORDER BY s1."block_number" DESC, s1."transaction_hash" DESC, t1."nonce", s1."from_address_hash" DESC, s1."to_address_hash" DESC
```

In general, the order should match the order of the subquery but the `nonce` part could slightly change the order of some transfers.

Reply:
yeah good catch, I completely missed the order here